source/change-streams/tests/change-streams.yml

```diff
@@ -0,0 +1,391 @@
+database_name: &database_name "change-stream-tests"
+collection_name_2: &collection_name_2 "test2"
+database_name_2: &database_name_2 "change-stream-tests-2"
+tests:
+  -
+    description: The watch helper must not throw a custom exception when executed against a single server topology, but instead depend on a server error
+    minServerVersion: "3.6.0"
+    target: collection
+    topology:
+      - single
+    changeStreamPipeline: []
+    changeStreamOptions: {}
+    operations: []
+    expectations: []
+    result:
+      error:
+        errmsg: The $changeStream stage is only supported on replica sets
+        code: 40573
+  -
+    description: "`$changeStream` must be the first stage in a change stream pipeline sent to the server"
+    minServerVersion: "3.6.0"
+    target: collection
+    topology: 
+      - replicaset
+    operations:
+      -
+        database: *database_name
+        collection: *collection_name
+        name: insertOne
+        arguments:
+          document:
+            x: 1
+    expectations:
+      - 
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            cursor: {}
+            pipeline:
+              - 
+                $changeStream:
+                  fullDocument: default
+          command_name: aggregate
+          database_name: *database_name
+    result: {}
+  -
+    description: The server returns change stream responses in the specified ``ServerResponse`` format
+    minServerVersion: "3.6.0"
+    target: collection
+    topology:
+      - replicaset
+    changeStreamPipeline: []
+    changeStreamOptions: {}
+    operations:
+      -
+        database: *database_name
+        collection: *collection_name
+        name: insertOne
+        arguments:
+          document:
+            x: 1
+    expectations: []
+    result:
+      success:
+        -
+          _id: "42"
+          documentKey: "42"
+          operationType: insert
+          ns:
+            db: *database_name
+            coll: *collection_name
+          fullDocument:
+            x:
+              $numberInt: "1"
+  -
+    description: Executing a `watch` helper on a Collection results in notifications for changes to the specified collection
+    minServerVersion: "3.6.0"
+    target: collection
+    topology:
+      - replicaset
+    changeStreamPipeline: []
+    changeStreamOptions: {}
+    operations:
+      -
+        database: *database_name
+        collection: *collection_name_2
+        name: insertOne
+        arguments:
+          document:
+            x: 1
+      -
+        database: *database_name_2
+        collection: *collection_name
+        name: insertOne
+        arguments:
+          document:
+            y: 2
+      -
+        database: *database_name
+        collection: *collection_name
+        name: insertOne
+        arguments:
+          document:
+            z: 3
+    expectations:
+      - 
+        command_started_event:
```

Comment:
Should APM not expect to see the inserts from the global client? For drivers that allow registering listeners globally, perhaps we need some language to mention that the APM listeners should only observe activity on the client under test (i.e. client executing the change stream).

Reply:
Good point.  I didn't realize some drivers had global listeners.  I'm not clear how a driver with a global listener could filter the inserts.  Perhaps just say that if a driver has a global listener, insert events should be ignored.

Reply:
Do any drivers _exclusively_ use global listeners? The language of the test runner says: "For each APM ``commandStarted`` events on ``client``", meaning only APM events on `client`. If that is not possible, then yeah, we should add a filter for specific APM events.

Reply:
PHP uses global listeners. This is a design necessity because our client objects and event listeners are both request-scoped, while the libmongoc clients are process-scoped and persisted across requests. Furthermore, since multiple PHP client objects could share the same libmongoc client within the same request (an edge case, but still possible), there isn't a good way to associate event listeners with individual PHP clients.

Simply stating that the listener should ignore events on the global client should be sufficient, though.